### PR TITLE
ci: tag the latest commit with the package major version

### PR DESCRIPTION
**Pull request by acazacu**
This PR changes the major version tag to be placed after the release version tag in the commit history.

### CODE REVIEW

File: .github/workflows/continuous-delivery.yaml [hunk 1]
```diff
@@ -43,8 +43,8 @@ jobs:
         name: Authenticate
         uses: actions/checkout@v2
         with:
-          fetch-depth: 1
           persist-credentials: true
+          ref: master
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:47]: I have trouble understanding why is it necessary. Did you test it on some other repositories?

**Comment by acazacu** [comment]
> This entire pipeline is only triggered on master right? 🤔

Quoting the checkout@v2 action README.md:

The branch, tag or SHA to checkout. When checking out the repository that triggered a workflow, this defaults to the reference or SHA for that event. Otherwise, defaults to `master`.

My understanding, if we don't pass a ref here, this will default to the commit SHA that triggered the workflow. Because the workflow pushes a commit (the release commit), the ref that we are tagging is one commit behind master.

You can see this in the tag list, in this repository: https://github.com/ridedott/release-me-action/tags

The major version tag is always before the latest release tag.

**Comment by merlinnot** [comment]
That is correct. However, it introduces some potential concurrency issues. I think we can try it, but please add a clear comment in the code.

**Comment by acazacu** [comment]
For the risks that it introduces, this PR is not worth implementing.

**Pull request closed without merging**